src/capstone_project_team_5/skill_detection.py

```diff
@@ -145,38 +147,239 @@ def scan_directory(directory: Path) -> None:
                         rel_path = str(item.relative_to(root)).replace("\\", "/")
 
                         # Detect tools and practices
-                        tools.update(SkillDetector._detect_tools(file_name, rel_path))
-                        practices.update(SkillDetector._detect_practices(file_name, rel_path))
-            except PermissionError:
-                # TODO: Add logging in later PR
+                        tools.update(SkillDetector._detect_tools_locally(file_name, rel_path))
+                        practices.update(
+                            SkillDetector._detect_practices_locally(file_name, rel_path)
+                        )
+            except (PermissionError, OSError, FileNotFoundError):
+                # Skip directories we can't access due to permissions or I/O errors
                 pass
 
         scan_directory(root)
         return tools, practices
 
+    @staticmethod
+    def _generate_directory_tree(
+        root: Path, prefix: str = "", max_depth: int = 5, current_depth: int = 0
+    ) -> str:
+        """
+        Generate a directory tree structure string for the project.
+
+        Args:
+            root: Root directory of the project
+            prefix: Prefix for indentation
+            max_depth: Maximum depth to traverse
+            current_depth: Current depth in the tree
+
+        Returns:
+            String representation of the directory tree
+        """
+        if current_depth >= max_depth:
+            return ""
+
+        tree_lines = []
+
+        try:
+            # Get all items in the directory
+            items = sorted(root.iterdir(), key=lambda x: (not x.is_dir(), x.name.lower()))
+
+            for item in items:
+                # Skip excluded directories
+                if item.is_dir() and SkillDetector._should_skip(item):
+                    continue
+
+                # Add item to tree
+                if item.is_dir():
+                    tree_lines.append(f"{prefix}{item.name}/")
+                    # Recursively add subdirectory contents
+                    subtree = SkillDetector._generate_directory_tree(
+                        item, prefix + "  ", max_depth, current_depth + 1
+                    )
+                    if subtree:
+                        tree_lines.append(subtree)
+                else:
+                    tree_lines.append(f"{prefix}{item.name}")
+
+        except (PermissionError, OSError, FileNotFoundError):
+            # Skip directories we can't access due to permissions or I/O errors
+            pass
+
+        return "\n".join(tree_lines)
+
+    @staticmethod
+    def _generate_llm_call_config(directory_tree: str) -> tuple[str, str]:
```

Comment:
The return type should probably be `tuple[str, str, float, int]`


```suggestion
    def _generate_llm_call_config(directory_tree: str) -> tuple[str, str, float, int]:
```